amoro-ams/src/main/java/org/apache/amoro/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -917,17 +955,23 @@ static boolean willNotRetain(
     }
   }
 
-  private static Literal<Long> getExpireTimestampLiteral(
+  private Literal<Long> getExpireTimestampLiteral(
       ContentFile<?> contentFile,
       Types.NestedField field,
       DateTimeFormatter formatter,
-      String numberDateFormatter) {
+      String numberDateFormatter,
+      Comparable<?> expireValue) {
     Type type = field.type();
     Object upperBound =
         Conversions.fromByteBuffer(type, contentFile.upperBounds().get(field.fieldId()));
     Literal<Long> literal = Literal.of(Long.MAX_VALUE);
+    if ("none".equals(table.properties().get("write.metadata.metrics.default"))) {
+      upperBound = null;
+    }
```

Comment:
@XBaith @czy006 I have added test cases, but without these three lines of code, two test cases fail. However, these three lines are currently unnecessary and are only used to solve the issue of the write.metadata.metrics.default configuration not being effective. I now need to have the test cases pass while removing these three lines of code.